src/ReplicatedStorage/Client/InstanceComponents/Misc/FallDamage.lua

```diff
@@ -13,8 +13,7 @@ function FallDamage.new(character: Model)
 	self._trove:Connect(self._humanoid.StateChanged, function(old, new)
 		if not (old == Enum.HumanoidStateType.Freefall and new == Enum.HumanoidStateType.Landed) then return end
 
-		local humanoidRootPart = character:WaitForChild("HumanoidRootPart")
-		local endVelocity = humanoidRootPart.AssemblyLinearVelocity.Magnitude
+		local endVelocity = self._humanoid.RootPart.AssemblyLinearVelocity.Magnitude
```

Comment:
You have the same issue as in the ocean damage check, you've gotta make sure this value is not `nil` since the Humanoid might not have a root part.

Reply:
I would also rename this to `endSpeed` since you're measuring the speed (magnitude of velocity)